genienlp/data_utils/iterator.py

```diff
@@ -100,6 +103,13 @@ def __next__(self):
                 longest_example = self.data_source[batch_of_indices[0]] # we sorted in descending order of length, so the first one is the longest
             else:
                 longest_example = new_example # this is the first element in batch, and therefore the longest
+            examples_size = self.batch_size_fn(longest_example)
+            if examples_size > self.batch_size:
+                logger.warning('Skipping an example larger than batch size. Consider increasing the batch size to avoid this warning')
+                self.last_batch_start_index += 1
+                i = self._get_next_batch_start_index()
```

Comment:
This might have to look around several times to find a short enough example.
You should just do `i += 1` and continue.